src/WebJobs.Extensions.DurableTask/AzureStorageAccountProvider.cs

```diff
@@ -44,7 +49,11 @@ public StorageAccountDetails GetStorageAccountDetails(string connectionName)
             AzureStorageAccountOptions account = connectionInfo.Get<AzureStorageAccountOptions>();
             if (account != null)
             {
-                TokenCredential credential = this.credentialFactory.Create(connectionInfo);
+                string key = !string.IsNullOrEmpty(account.AccountName) ? account.AccountName : account.BlobServiceUri.ToString();
+
+                TokenCredential credential = this.cachedTokenCredentials.GetOrAdd(
+                    key,
+                    attr => this.credentialFactory.Create(connectionInfo));
```

Comment:
Doesn't this token eventually expire? We seem to be storing it forever, but doesn't it get invalidated on the server side after a long enough period? Shouldn't we refresh it sometimes?

Reply:
Tagging @jviau because from what I remember he addressed this in a weekly sync when we originally discussed this memory leak, but let me know if I am remembering incorrectly. @jviau, do you know if there's any additional work we have to do to handle the token refresh?

Reply:
`TokenCredentials` are long lived objects which refresh their token internally

Reply:
Thank you @jviau. Are you saying that _the background auto refresh task leaks memory_, am I reading this correctly? Or the leak happens only when a new instance of `TokenCredential` is created?

Reply:
It is the number of background tasks that adds up with each creation of the credential. The tasks logic itself does not leak memory but the task is itself leaked.

Reply:
Ok, I understand now, thank you.